ui/src/components/Views/Configs/MountPointForm.tsx

```diff
@@ -1,137 +1,187 @@
+import { ExecResult } from '@docker/extension-api-client-types/dist/v1';
 import {
-  Box,
   Button,
   CircularProgress,
   Dialog,
   DialogActions,
   DialogContent,
+  Divider,
   FormControl,
   MenuItem,
+  Paper,
   Select,
   Typography,
 } from '@mui/material';
 import React, { ReactElement, useEffect, useState } from 'react';
-import { LATEST_IMAGE } from '../../../constants';
-import { useDDClient, useMountPoint } from '../../../services';
-import { DockerImage } from '../../../types';
-import { DownloadProgress } from '../../Feedback';
+import {
+  getOSsFromBinary,
+  getUsersFromBinaryUnix,
+  getUsersFromBinaryWindows,
+  useDDClient,
+  useMountPoint,
+} from '../../../services';
+
+const ShrinkedCircularProgress = (): ReactElement =>  <CircularProgress size={20} sx={{ margin: 1}}/>;
 
 export const MountPointForm = (): ReactElement => {
+
   const [userState, setUserState] = useState({ loading: false, selectedUser: '', users: [] });
-  const [hasLocalImage, setHasLocalImage] = useState({ checking: true, isPresent: false });
-  const [isPullingImage, setIsPullingImage] = useState(false);
-  const [triggerImageCheck, setTriggerImageCheck] = useState(false);
+  const [osState, setOsState] = useState({ loading: false, selectedOS: '', OSs: [] });
+  const [triggerUserCheck, setTriggerUserCheck] = useState(false);
 
-  const { setMountPointUser } = useMountPoint();
+  const { setMountPointData } = useMountPoint();
   const ddClient = useDDClient();
 
-  const checkHomeDir = async () => {
-    setUserState({ loading: true, selectedUser: userState.selectedUser, users: userState.users });
-    const path = ddClient.host.platform === 'darwin' ? 'Users' : 'home';
-    const res = await ddClient.docker.cli.exec('run',
-      ['--rm', '--entrypoint=', '-v', `/${path}:/users`, 'localstack/localstack', 'ls', '/users']);
-
-    if (res.stderr !== '' || res.stdout === '') {
-      ddClient.desktopUI.toast.error(`Error while locating users: ${res.stderr}\n using /tmp as mount point`);
-      setUserState({ loading: false, selectedUser: 'tmp', users: ['tmp'] });
-      setMountPointUser('tmp');
+  const getMountPointPath = (): string => {
+    if(ddClient.host.platform === 'darwin'){
+      return `/Users/${userState.selectedUser || 'loading...'}/Library/Caches/localstack/volume`;
```

Comment:
We should make sure to only use this string (including the string `loading...`) for display purposes, and not as an actual mount point. (haven't fully checked whether that's the case)

Reply:
this is only for display :smile: